lib/client.js

```diff
@@ -256,9 +256,16 @@ Client.prototype._attachApi = function () {
              *  @param {Object} response - response from swagger
              */
             function processResponse (response) {
-              var result = response.data.toString('utf-8');
-              if (respType !== null && result) {
-                result = JSON.parse(result);
+              var result;
+
+              if (respType === 'binary') {
+                result = response.data;
```

Comment:
instead of making changes to knownTypes and adding an object that just parses the result into a node.js Buffer, why not just do it here? 

```
if (respType === 'binary') {
  result = Buffer.from(response.data);
} else { ... }
```